llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -966,6 +995,39 @@ Error MCCASBuilder::buildFragments() {
       continue;
     }
 
+    if (Asm.getContext().getObjectFileInfo()->getDwarfLineSection() == &Sec) {
+      SmallVector<char> DebugLineContent;
+      for (const MCFragment &F : Sec) {
```

Comment:
Does this section have more than one fragment? IIRC the "debug info" section only has a single fragment

Reply:
> In general, the LLVM's programmer manual prescribes [always favouring](https://llvm.org/docs/ProgrammersManual.html#llvm-adt-smallvector-h) SmallVector<T, 0> over std::vector.

TIL!

Reply:
> Regarding the "SmallSize" default, do we have any good metrics for what a type DebugLine section would look like for a "Small" function?

This is probably a microoptimization that is largely irrelevant, but we could gather statistics from e.g., a build of Clang if we wanted to. But really, size 0 seems fine, too.

Reply:
@felipepiovezan I am pretty sure that the Debug Info section can have multiple fragments, and you should not assume that it will always be one fragment. You could have multiple data fragments as well.